lib/core/state/replication.js

```diff
@@ -0,0 +1,46 @@
+const MongoClient = require('../clients/mongodb-client')
+const ReplicationCommands = require('../commands/replication-commands')
+const { REPLICATION_STATUS, MongoServerState } = require('./mongo-server')
+const logger = require('../../helpers/logger')
+
+const REPLICA_SET_STATUS = {
+  UNKNOWN: 0,
+  UNINITIALIZED: 1
+}
+
+class ReplicaSetState {
+  constructor(configuration) {
+    this.hostnames = configuration.hostnames
```

Comment:
I would actually prefer to see something like an array of  `MongoServerState`'s? Is there a reason why you didn't  do this?

Reply:
Went for an implementation like that